File: 2021년/7월/0705 로또.py
# 백준_6603_로또_백트래킹_실버 2

import sys
input = sys.stdin.readline


def solve(k):
    if len(nums) == 6:
        result.append(tuple(sorted(nums)))
        return

    for i in range(k):
        if visited[i] == 0:
            nums.append(lst[i])
            visited[i] = 1
            solve(k)
            visited[i] = 0
            nums.pop()


while 1:
    lst = list(map(int, input().split()))
    k = lst[0]
    if k == 0:
        break
    lst = lst[1:]
    visited = [0]*k
    nums = []
    result = []

    solve(k)

    for num in sorted(list(set(result))):
        print(' '.join(map(str, num)))

    print()

File: 2021년/7월/0705 로또.cpp
// 백준_6603_로또_백트래킹_실버 2

#include <iostream>
#include <vector>

using namespace std;

int k;
int nums[13];
int result[6];

void solve(int index, int depth)
{
	if (depth == 6)
	{
		for(int i =0;i<6;i++) cout << result[i] << ' ';
		cout << endl;
		return;
	}
	
	
	for(int i=index;i<k;i++)
	{
		result[depth] = nums[i];
		solve(i+1, depth+1);
	}
}

int main()
{
	while(1)
	{
		cin >> k;
		if (k == 0) break;
		for (int i = 0;i<k;i++) cin >> nums[i];
		solve(0,0);
		cout << endl;
	}
	
}
